uefi-mkconfig

```diff
@@ -198,58 +157,109 @@ main () {
 
 	# Verify kernel commands
         for check_config in $kernel_commands; do
-
                 # Strip initrd=* from config file
                 [[ "$check_config" != *"initrd="* ]] && strip_kernel_commands="$strip_kernel_commands $check_config"
                 # Verify if "root=" is present
                 [[ "$check_config" == "root="* ]] && valid_kernel_commands=1
-
         done
 	kernel_commands="$strip_kernel_commands"
 
         [[ -z $valid_kernel_commands ]] && ewarn "Warning, kernel command \"root=\" is missing from loaded configuration!"
+	
+	# Get initial state of UEFI entries
+	initial_uefi_state="$(efibootmgr -u)"
+	
+	# Wipe all managed entries for regeneration
+	wipe_entries
+	
+	partition_mounts="${mounted_efi_partitions// "$efi_parttype"/}"
+	
+	# Find all .efi files on this partition and split them to old and new
+	for prefix in $kernel_prefixes; do
+		find_results_new="$(find $partition_mounts -name "$prefix*.efi" ! -name "$prefix*-old.efi" -printf "%f\n")"
+		find_results_old="$(find $partition_mounts -name "$prefix*-old.efi" -printf "%f\n")"
+		[[ -n "$find_results_new" ]] && kernel_images_new+="$find_results_new "
+		[[ -n "$find_results_old" ]] && kernel_images_old+="$find_results_old "
+	done
+
+	# Sort kernel images
+	kernel_images="$(sort -uV <<< "$kernel_images_old") $(sort -uV <<< "$kernel_images_new")"
+
+	# Add path back to the kernel image
+	for kernel_image in $kernel_images; do
+		# Variable partition_mounts in this case can't be inside of ""
+        	partition_efis+="$(find $partition_mounts -name "$kernel_image") "
+	done
+	
+	# Move backup entries to the beginning of the list
+	for efi_file in $partition_efis; do
+		[[ -f "${efi_file}.uefibackup" ]] && partition_efis="${efi_file}.uefibackup ${partition_efis}"
+	done
+	
+	for efi_file in $partition_efis; do
+		# Get partition mount
+		partition_mount="${efi_file/\/${efi_file#\/*\/}/}"
+
+		# Get partition
+		partition=$(mount | grep "$partition_mount ")
+		partition=${partition// */}
+		partition=${partition//*\//}
 		
-	# Clear old entries for regeneration
-	## 256..512 is because entry IDs are actually in hexadecimal format
-	for entry_number in {256..512}; do
-		# Wipe only boot entries with UMC stamp
-		if [[ "$(efibootmgr -u | grep Boot$(printf %04X $entry_number))" == *"UMC "* ]]; then
-			efibootmgr -q -B -b $(printf %04X $entry_number)
+		# Skip this kernel version if ignore file is found
+		if [[ -f "$efi_file.ignore" ]]; then
+			ewarn "Ignoring \"$efi_file\" on \"$partition\""
+			continue
 		fi
-	done
 
-	for partition in $mounted_efi_partitions; do
+		# Get partition label
+		partition_label="$(lsblk "/dev/$partition" -lno PARTLABEL)"
+		## In case PARTLABEL isn't set, use UUID
+		[[ -z "$partition_label" ]] && partition_label="$(lsblk "/dev/$partition" -lno UUID)"
```

Comment:
should we use the PARTUUID instead since that is what the old code used?

Reply:
In this case I think UUID is sufficient. This is just for labeling the entry so the output would look like this:

```
Boot01FF* UMCB 6.9.6-gentoo-dist on 4813-F852
```

PARTUUID is too long for this use case. But of course I could use just first 8 characters.

Reply:
> PARTUUID is too long for this use case. 

That makes sense